internal/services/task_status_listener.go

```diff
@@ -121,8 +123,16 @@ func (i *TaskStatusListener) processSmartcarPollStatusEvent(event *shared.CloudE
 	}
 
 	udai.Status = models.UserDeviceAPIIntegrationStatusAuthenticationFailure
-	_, err = udai.Update(context.Background(), i.db().Writer, boil.Infer())
-	return err
+	if _, err = udai.Update(context.Background(), i.db().Writer, boil.Infer()); err != nil {
+		i.log.Err(err).Str("userDeviceID", userDeviceID).Str("integrationID", integrationID).Msg("failed up update user device api integration with failure status")
```

Comment:
Should we be logging or returning the error here.

Reply:
my sense is we should log the error here but keep going. if we aren't able to update the db then the devices api db is out of sync but I don't think that needs to prevent us from sending an update to customer io

Reply:
👍🏾  I can get behind that since we just log on error any way

Reply:
Our error handling when Kafka messages fail to process is... it's not good. I don't see a need to _change_ the behavior like we're doing here but